File: solutions/lua/collatz-conjecture/1/collatz-conjecture.lua
local function collatz(n, s)
    assert(n > 0, 'Only positive numbers are allowed')
    s = s or 0
    return n == 1 and s or collatz(n % 2 == 0 and n >> 1 or 3 * n + 1, s + 1)
end

return collatz

File: solutions/php/collatz-conjecture/1/CollatzConjecture.php
<?php


declare(strict_types=1);

function steps(int $n): int
{
    if ($n < 1) {
        throw new InvalidArgumentException('Only positive numbers are allowed');
    }
    $steps = 0;
    while($n !== 1) {
        $steps++;
        if ($n % 2 === 0) {
            $n = $n / 2;
            continue;
        }
        $n = (3 * $n) + 1;
    }

    return $steps;
}